File: python_version/Owca.py
from Zwierze import Zwierze

class Owca(Zwierze):
    def __init__(self, x, y, swiat):
        super().__init__(4, 4, x, y, swiat)

    def rysowanie(self):
        return 'O'

    def get_nazwa(self):
        return "Owca"

    def stworz_potomka(self, x, y):
        return Owca(x, y, self.swiat)

    def czy_ten_sam_gatunek(self, inny):
        return isinstance(inny, Owca)

File: java_version/Owca.java
public class Owca extends Zwierze {
    public Owca(int x, int y, Swiat swiat) {
        super(4, 4, x, y, swiat); // siła 4, inicjatywa 4
    }

    @Override
    public char rysowanie() {
        return 'O';
    }

    @Override
    public String getNazwa() {
        return "Owca";
    }

    @Override
    public Organizm stworzPotomka(int nowyX, int nowyY) {
        return new Owca(nowyX, nowyY, swiat);
    }

    @Override
    public boolean czyTenSamGatunek(Organizm inny) {
        return inny instanceof Owca;
    }
}
